modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php

```diff
@@ -61,26 +71,74 @@ private function getDatabaseConnectionCapableOfDataLoad() {
   }
 
   /**
-   * Private.
+   * Properly escape and format the supplied list of column names.
+   *
+   * @param string[] $columns
+   *   List of column names.
+   *
+   * @return array
+   *   List of sanitized table headers keyed by original column names.
    */
-  private function cleanHeaders($headers) {
-    $row = [];
-    foreach ($headers as $field) {
-      $new = preg_replace("/[^A-Za-z0-9_ ]/", '', $field);
-      $new = trim($new);
-      $new = strtolower($new);
-      $new = str_replace(" ", "_", $new);
-
-      $mysqlMaxColLength = 64;
-      if (strlen($new) > $mysqlMaxColLength) {
-        $strings = str_split($new, $mysqlMaxColLength - 5);
-        $token = substr(md5($field), 0, 4);
-        $new = $strings[0] . "_{$token}";
-      }
-
-      $row[] = $new;
+  private function generateTableHeaders(array $columns): array {
+    return array_replace([], ...array_map(function ($column) {
+      // Sanitize the supplied table header to generate a unique column name.
+      $header = $this->sanitizeHeader($column);
+      // Prepend "d_" to the table column name to prevent numeric column name
+      // parsing issues.
+      //
+      // @see https://github.com/GetDKAN/dkan/issues/3606
+      $header = 'd_' . $header;
```

Comment:
Will this prefix show up in the api & front-end? I wonder if we should be trying to remove this in column names for reads/queries, since it's not really relevant to end users? I also wonder if the regular importer should do add this prefix too for consistency?

Reply:
I should have looked at https://github.com/GetDKAN/datastore/pull/35 first! I think that addresses this for the most part, except perhaps that we may still want to update the regular importer with the same heuristic?

Reply:
Yes, agreed. The regular importer also seems to be afflicted by the same bug as well.